sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala

```
@@ -22,18 +22,87 @@ import java.util.Properties
+ * about calculating the original path.  The MULTITHREADED reader supports input_file_name
+ * so it handles calculating the original file path in the case of the conver time algorithm.
+ * In order to do the replacement at task time and to output the original path for convert
+ * time, we need to have a mapping of the original scheme to the alluxio scheme. This has been
+ * made a parameter to many of the readers. With auto mount and task time replacement,
+ * we make a pass through the files on the driver side in GpuFileSourceScanExec in order to
+ * do the mounting before the tasks try to access alluxio.
+ * Note that Delta Lake uses the input_file_name functionality to do things like
+ * Updates and Deletes and will fail if the path has the alluxio:// in it.
+ *
+ * Below we support 2 configs to turn on Alluxio, we have the automount which uses a regex
+ * to replace paths and then we have the config that specifies direct paths to replace and
+ * user has to manually mount those.
+ */
 object AlluxioUtils extends Logging {
   private val checkedAlluxioPath = scala.collection.mutable.HashSet[String]()
+  private val ALLUXIO_SCHEME = "alluxio://"
+  private val mountedBuckets: scala.collection.mutable.Map[String, String] =
+    scala.collection.mutable.Map()
+  private var alluxioCmd: Seq[String] = null
+  private var alluxioMasterHost: Option[String] = None
+  private var alluxioPathsToReplaceMap: Option[Map[String, String]] = None
+  private var alluxioHome: String = "/opt/alluxio-2.8.0"
+  private var isInit: Boolean = false
+
+  def checkAlluxioNotSupported(rapidsConf: RapidsConf): Unit = {
+    if (rapidsConf.isParquetPerFileReadEnabled &&
+      (rapidsConf.getAlluxioAutoMountEnabled || rapidsConf.getAlluxioPathsToReplace.isDefined)) {
+      throw new IllegalArgumentException("Alluxio is currently not supported with the PERFILE " +
+        "reader, please use one of the other reader types.")
```

would be nice if this said what reader to use? i.e. (MULTITHREADED or COALESCING)

---

should we silently fallback to MULTITHREADED or COALESCING by default? and warn about it ... I don't know what the right answer is.

---

I'd rather just error because PERFILE isn't the default so user likely set it for specific reason like working around another problem.